iceoryx_hoofs/include/iceoryx_hoofs/error_handling/error_handler.hpp

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2019 - 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2020 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_ERROR_HANDLING_ERROR_HANDLER_HPP
+#define IOX_HOOFS_ERROR_HANDLING_ERROR_HANDLER_HPP
+
+#include "iceoryx_hoofs/cxx/generic_raii.hpp"
+#include "iceoryx_hoofs/log/logger.hpp"
+#include "iceoryx_hoofs/log/logging.hpp"
+#include "iceoryx_hoofs/log/logmanager.hpp"
+
+#include <functional>
+#include <iostream>
+
+namespace iox
+{
+constexpr uint16_t HOOFS_MODULE_IDENTIFIER{1};
+constexpr uint16_t POSH_MODULE_IDENTIFIER{2};
+constexpr uint16_t C_BINDING_MODULE_IDENTIFIER{3};
+constexpr uint16_t USER_DEFINED_MODULE_IDENTIFIER{256};
```

Comment:
Well, this is about preventing the user to use arbitrary numbers by giving them a process to follow. If they use arbitrary numbers it is already too late and not in our control anymore. If we specify it now, we can tell the user `it's your fault™`

Reply:
TBH I don't renaming the variable will prevent users from using arbitrary numbers. If we want to prevent misuse here we would need a proper module registration like used in other logging and error handling frameworks. For now, I would keep it simple and stupid. No hard feelings on the renaming, though.

Reply:
Preventing was maybe a bit exaggerated. It's more like making it obvious how to use/not use it. Having a `USER_DEFINED_MODULE_IDENTIFIER` is already quite good in this regard. Just wanted to be sure that we are on the same page and talk about the same stuff.

What do you think about a comment above `USER_DEFINED_MODULE_IDENTIFIER` to clarify that every number larger than that can be used by external users?